tests/Integration/Momento.Sdk.Tests/SetTest.cs

```diff
@@ -572,6 +577,63 @@ public async Task SetFetchAsync_UsesCachedStringSet_HappyPath()
         var set2 = hitResponse.ValueSetString;
         Assert.Same(set1, set2);
     }
+    
+    
+    [Theory]
+    [InlineData(null, "my-set")]
+    [InlineData("cache", null)]
+    public async Task SetSampleAsync_NullChecks_IsError(string cacheName, string setName)
+    {
+        CacheSetSampleResponse response = await client.SetSampleAsync(cacheName, setName, 100);
+        Assert.True(response is CacheSetSampleResponse.Error, $"Unexpected response: {response}");
+        Assert.Equal(MomentoErrorCode.INVALID_ARGUMENT_ERROR, ((CacheSetSampleResponse.Error)response).ErrorCode);
+    }
+
+    [Fact]
+    public async Task SetSampleAsync_Missing_HappyPath()
+    {
+        var setName = Utils.NewGuidString();
+        CacheSetSampleResponse response = await client.SetSampleAsync(cacheName, setName, 100);
+        Assert.True(response is CacheSetSampleResponse.Miss, $"Unexpected response: {response}");
+    }
+
+    [Fact]
+    public async Task SetSampleAsync_UsesCachedStringSet_HappyPath()
+    {
+        var setName = Utils.NewGuidString();
+        var allValues = new HashSet<String> { "jalapeno", "habanero", "serrano", "poblano" };
+        CacheSetAddElementsResponse setResponse = await client.SetAddElementsAsync(cacheName, setName, allValues);
+        Assert.True(setResponse is CacheSetAddElementsResponse.Success, $"Unexpected response: {setResponse}");
+
+        CacheSetSampleResponse allElementsResponse = await client.SetSampleAsync(cacheName, setName, allValues.Count);
+        Assert.True(allElementsResponse is CacheSetSampleResponse.Hit, $"Unexpected response: {allElementsResponse}");
+        var allElementsHitValues = ((CacheSetSampleResponse.Hit)allElementsResponse).ValueSetString;
+        Assert.True(allValues.SetEquals(allElementsHitValues), $"Expected sample with with limit matching set size to return the entire set; expected ({String.Join(", ", allValues)}), got ({String.Join(", ", allElementsHitValues)})");
+        
+        CacheSetSampleResponse limitGreaterThanSetSizeResponse = await client.SetSampleAsync(cacheName, setName, 1000);
+        Assert.True(limitGreaterThanSetSizeResponse is CacheSetSampleResponse.Hit, $"Unexpected response: {limitGreaterThanSetSizeResponse}");
+        var limitGreaterThanSetSizeHitValues = ((CacheSetSampleResponse.Hit)limitGreaterThanSetSizeResponse).ValueSetString;
+        Assert.True(allValues.SetEquals(limitGreaterThanSetSizeHitValues), $"Expected sample with with limit greater than set size to return the entire set; expected ({String.Join(", ", allValues)}), got ({String.Join(", ", limitGreaterThanSetSizeHitValues)})");
+        
+        CacheSetSampleResponse limitZeroResponse = await client.SetSampleAsync(cacheName, setName, 0);
```

Comment:
Same comment here re: the `ArgumentNotNull` check on the limit. This is returning a miss as opposed to an error.